### PR TITLE
Fix rolling upgrade wait condition

### PR DESCRIPTION
**Pull request by mgencur**
* take into account both cases when we installed previous version via
old CR and also via new CR

### CODE REVIEW

File: test/lib.bash [hunk 1]
```diff
@@ -173,9 +173,13 @@ function run_knative_serving_rolling_upgrade_tests {
...
     else
       approve_csv "$upgrade_to" || return 1
-      # Both knativeserving CRs should be updated now
-      timeout 900 '[[ ! ( $(oc get knativeserving.serving.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
-      timeout 900 '[[ ! ( $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
+      # If we used old CR for installing Serverless both knativeserving CRs should be updated now
+      if oc get knativeserving.serving.knative.dev knative-serving -n "$SERVING_NAMESPACE" >/dev/null 2>&1; then
+        timeout 900 '[[ ! ( $(oc get knativeserving.serving.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
+        timeout 900 '[[ ! ( $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
+      else
```
Comment (markusthoemmes) [inline test/lib.bash]: I think you can drop the else block and always run the check for the new group. A bit less duplication.

Comment (mgencur) [reply test/lib.bash]: What if we installed Serverless via the new CR? Then it wouldn't wait at all?

Comment (mgencur) [reply test/lib.bash]: Oh. Sorry. Got it.

Comment (mgencur) [reply test/lib.bash]: Pushed.

**Comment by markusthoemmes** [comment]
Should this go against master as well?

**Comment by mgencur** [comment]
Yes. I'll send one against master as well.

**Pull request merged**